### PR TITLE
Groud work for keystone v3 changes

### PR DESCRIPTION
**Pull request by rhafer**
This pull request should contain the minimal changes required for keystone v3 support. It's a subset of https://github.com/crowbar/barclamp-keystone/pull/277 and should help to speed up review and merging of the changes to the other barclamps independent of  the bigger changes in: https://github.com/crowbar/barclamp-keystone/pull/277

It intention is to have this merged first (v2 functionality should keep working without changes, v3 will only work after the other barclamp have been updated as well), then merge the PRs on the other barclamps and finally merge the rest of the keystone changes in #277.

### CODE REVIEW

**Comment by vuntz** [comment]
+1

File: chef/cookbooks/keystone/templates/default/openrc.erb [hunk 1]
```diff
@@ -2,6 +2,12 @@
 export OS_USERNAME='<%= @keystone_settings['admin_user'] %>'
 export OS_PASSWORD='<%= @keystone_settings['admin_password'] %>'
 export OS_TENANT_NAME='<%= @keystone_settings['default_tenant'] %>'
+export OS_IDENTITY_API_VERSION='<%= @keystone_settings['api_version'] %>'
+<% if @keystone_settings['api_version'] != '2.0' %>
+export OS_USER_DOMAIN_NAME='Default'
+export OS_PROJECT_DOMAIN_NAME='Default'
+export OS_AUTH_VERSION='<%= @keystone_settings['api_version'] %>'
```
Comment (dirkmueller) [inline chef/cookbooks/keystone/templates/default/openrc.erb]: is this intentionally inside the <% if> block? the OS_IDENTITY_API_VERSION is outside

File: chef/cookbooks/keystone/libraries/helpers.rb [hunk 2]
```diff
@@ -36,7 +36,11 @@ def self.keystone_settings(current_node, cookbook_name)
 
       @keystone_settings ||= Hash.new
       @keystone_settings[cookbook_name] = {
-        "api_version" => node[:keystone][:api][:version].sub(/^v/, ""),
+        "api_version" => node[:keystone][:api][:version],
+        # This is somehwat ugly but the Juno keystonemiddleware expects the
+        # version to be a "v3.0" for the v3 API instead of the "v3" or "3" that
+        # is used everywhere else.
+        "api_version_for_middleware" => "v#{node[:keystone][:api][:version].to_f}",
```
Comment (dirkmueller) [inline chef/cookbooks/keystone/libraries/helpers.rb]: is .to_f always generating a one digit suffix? I suggest

```ruby
 "v%.1f" % node[:keystone][:api][:version]
```

which makes this more explicit IMHO

File: chef/cookbooks/keystone/templates/default/openrc.erb (hunk 1 above)
Comment (rhafer) [inline chef/cookbooks/keystone/templates/default/openrc.erb]: > is this intentionally inside the <% if> block? the OS_IDENTITY_API_VERSION is outside

No. Fixed now.

File: chef/cookbooks/keystone/libraries/helpers.rb (hunk 2 above)
Comment (rhafer) [inline chef/cookbooks/keystone/libraries/helpers.rb]: Ok, adapted.

**Comment by dirkmueller** [comment]
+1

**Pull request merged**